lib/storage/storage.go

```diff
@@ -673,7 +675,7 @@ var freeDiskSpaceLimitBytes uint64
 
 // IsReadOnly returns information is storage in read only mode
 func (s *Storage) IsReadOnly() bool {
-	return s.isReadOnly.Load()
+	return s.isReadOnly.Load() && s.cannotWrite.Load()
```

Comment:
return s.isReadOnly.Load() || s.cannotWrite.Load()

Reply:
It seems unnecessary to introduce a new variable `cannotWrite` here; the readonly watchdog can update `isReadOnly` to `false` when it detects a disk issue.